src/tools/coverage-dump/src/covfun.rs

```diff
@@ -159,11 +162,30 @@ impl<'a> Parser<'a> {
             match high {
                 0 => unreachable!("zero kind should have already been handled as a code mapping"),
                 2 => Ok(MappingKind::Skip),
-                4 => {
+                4 | 6 => {
                     let r#true = self.read_simple_term()?;
                     let r#false = self.read_simple_term()?;
-                    Ok(MappingKind::Branch { r#true, r#false })
+                    if high == 6 {
+                        let condition_id = self.read_uleb128_u32()?;
+                        let true_next_id = self.read_uleb128_u32()?;
+                        let false_next_id = self.read_uleb128_u32()?;
+                        Ok(MappingKind::MCDCBranch {
+                            r#true,
+                            r#false,
+                            condition_id,
+                            true_next_id,
+                            false_next_id,
+                        })
+                    } else {
+                        Ok(MappingKind::Branch { r#true, r#false })
+                    }
```

Comment:
4 and 6 should be separate match arms (with 6 coming after 5).

It's fine for 4 and 6 to both read separate copies of their own `r#true` and `r#false`.